backend/lib/plugins/socket.js

```diff
@@ -438,8 +438,11 @@ function onSocketConnect(socket) {
   });
 
   socket.on("CLEAR_NOTIFICATION", async (data) => {
+    const { userId } = socket;
     const { notificationId } = data;
-    await Notification.findByIdAndUpdate(notificationId, { isCleared: true })
+    const notification = await Notification.findById(notificationId);
+    if(notification.receiver === userId) 
```

Comment:
Does this work? because `notification.receiver` is `ObjectId`, and `userId` is `String`, maybe you will need to add `notification.receiver.toString()`, but if you tested it and it works then that's fine.

Reply:
it works in testing, I'm pretty sure mongo automatically converts to string when comparing. I have it that way in my own projects and it never seems to fail. But I'll add `toString` to the next commit just to be sure.